lib/linguist/strategy/modeline.rb

```diff
@@ -22,8 +22,13 @@ def self.call(blob, _ = nil)
       #
       # Returns a String or nil
       def self.modeline(data)
-        match = data.match(EmacsModeline) || data.match(VimModeline)
-        match[1] if match
+        match = data.match(EmacsModeline)
+        if match
+          match[4]
+        else
+          match = data.match(VimModeline)
+          match[1] if match
+        end
```

Comment:
I think if you use `?:` for the other capturing groups you wouldn't have to change this part.